pkg/ibm/ibmz_helpers.go

```diff
@@ -2,106 +2,149 @@ package ibm
+// createInstanceName returns a unique instance name in the
+// format  <instance_tag>-<instance_id> where the 'instance_id'
+// is a 20 character universally unique ID generated using the
+// md5 cryptographic hash function.
+//
+// Used in for Both IBM System Z & IBM Power PC.
+func createInstanceName(instanceTag string) (string, error) {
+	binary, err := uuid.New().MarshalBinary()
+	if err != nil {
+		return "", fmt.Errorf("failed to create a UUID for the instance name: %w", err)
+	}
+	// #nosec is added to bypass the golang security scan since the cryptographic
+	// strength doesn't matter here
+	md5EncodedBinary := md5.New().Sum(binary) //#nosec
+	md5EncodedString := base64.URLEncoding.EncodeToString(md5EncodedBinary)[0:20]
+	instanceId := strings.Replace(strings.ToLower(md5EncodedString), "_", "-", -1)
+	return fmt.Sprintf("%s-%sx", instanceTag, instanceId), nil
+}
+
+// checkIfIpIsLive tries to connect to the IP address ip. An
+// error is returned if ip couldn't be reached.
+//
+// Used in for Both IBM System Z & IBM Power PC.
+func checkIfIpIsLive(ctx context.Context, ip string) error {
+	log := logr.FromContextOrDiscard(ctx)
+	log.Info(fmt.Sprintf("checking if IP address %s is live", ip))
+
+	server, _ := net.ResolveTCPAddr("tcp", ip+":22")
+	conn, err := net.DialTimeout(server.Network(), server.String(), 5*time.Second)
+	if err != nil {
+		log.Info("failed to connect to IBM host " + ip)
+		return err
+	}
+	defer conn.Close()
+	return nil
+
+}
+
+func ptr[V any](s V) *V {
+	return &s
+}
+
+// lookUpSubnet looks for ibmz's subnet in the provided IBM Virtual Private Cloud service's API.
 // Either the corresponding subnet or an error with a nil object is returned.
-func (r IBMZDynamicConfig) lookupSubnet(vpcService *vpcv1.VpcV1) (*vpcv1.Subnet, error) {
+func (ibmz IBMZDynamicConfig) lookUpSubnet(vpcService *vpcv1.VpcV1) (*vpcv1.Subnet, error) {
 	subnets, _, err := vpcService.ListSubnets(&vpcv1.ListSubnetsOptions{})
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to list VPC subnets: %w", err)
 	}
 
 	var subnet *vpcv1.Subnet
 	for i := range subnets.Subnets {
-		if *subnets.Subnets[i].Name == r.Subnet {
+		if *subnets.Subnets[i].Name == ibmz.Subnet {
 			subnet = &subnets.Subnets[i]
 			break
 		}
 	}
 	if subnet == nil {
-		return nil, fmt.Errorf("failed to find subnet %s", r.Subnet)
+		return nil, fmt.Errorf("failed to find subnet %s", ibmz.Subnet)
 	}
 	return subnet, nil
 }
 
-// lookupSSHKey looks for ibmz's SSH key in the provided IBM Virtual Private Cloud service's API.
+// lookUpSshKey looks for ibmz's SSH key in the provided IBM Virtual Private Cloud service's API.
 // Either the corresponding key or an error with a nil object is returned.
-func (r IBMZDynamicConfig) lookupSSHKey(vpcService *vpcv1.VpcV1) (*vpcv1.Key, error) {
+func (ibmz IBMZDynamicConfig) lookUpSshKey(vpcService *vpcv1.VpcV1) (*vpcv1.Key, error) {
```

Comment:
Just a nitpick, but it is reccomended style to have a method receiver names lot longer than 2 chars, like `i` or `iz` 
https://go.dev/wiki/CodeReviewComments#receiver-names